cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -147,6 +147,7 @@ protected void startUp() throws Exception {
     // Run http service on random port
     httpService = new CommonNettyHttpServiceBuilder(configuration)
       .setHost(hostname.getCanonicalHostName())
+      .setPort(configuration.getInt(Constants.AppFabric.SERVER_PORT))
```

Comment:
this will throw a NullPointerException if the configuration does not contain Constants.AppFabric.SERVER_PORT. A default value of 0 needs to be added to cdap-common/src/main/resources/cdap-default.xml.

And the same for all the other newly introduced port configs.